File: Test/AdventureWorksFunctionalModel/Mapping/PersonMap.cs
using System.Data.Entity.ModelConfiguration;
using AW.Types;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
#pragma warning disable 8602

namespace AW.Mapping {
    public class PersonMap : EntityTypeConfiguration<Person> {
        public PersonMap() {
            // Primary Key
            HasKey(t => t.BusinessEntityID);

            Property(t => t.Title)
                .HasMaxLength(8);

            Property(t => t.FirstName)
                .IsRequired()
                .HasMaxLength(50);

            Property(t => t.MiddleName)
                .HasMaxLength(50);

            Property(t => t.LastName)
                .IsRequired()
                .HasMaxLength(50);

            Property(t => t.Suffix)
                .HasMaxLength(10);

            // Table & Column Mappings
            ToTable("Person", "Person");
            Property(t => t.BusinessEntityID).HasColumnName("BusinessEntityID");
            Property(t => t.PersonType).HasColumnName("PersonType");
            Property(t => t.NameStyle).HasColumnName("NameStyle");
            Property(t => t.Title).HasColumnName("Title");
            Property(t => t.FirstName).HasColumnName("FirstName");
            Property(t => t.MiddleName).HasColumnName("MiddleName");
            Property(t => t.LastName).HasColumnName("LastName");
            Property(t => t.Suffix).HasColumnName("Suffix");
            Property(t => t.EmailPromotion).HasColumnName("EmailPromotion");
            Property(t => t.AdditionalContactInfo).HasColumnName("AdditionalContactInfo");
            Property(t => t.rowguid).HasColumnName("rowguid");
            Property(t => t.ModifiedDate).HasColumnName("ModifiedDate");

            HasOptional(t => t.Employee).WithRequired(t => t.PersonDetails);
            HasOptional(t => t.Password).WithRequired(pw => pw.Person);
        }
    }

    public static partial class Mapper {
        public static void Map(this EntityTypeBuilder<Person> builder) {
            builder.Property(t => t.Title)
                   .HasMaxLength(8);

            builder.Property(t => t.FirstName)
                   .IsRequired()
                   .HasMaxLength(50);

            builder.Property(t => t.MiddleName)
                   .HasMaxLength(50);

            builder.Property(t => t.LastName)
                   .IsRequired()
                   .HasMaxLength(50);

            builder.Property(t => t.Suffix)
                   .HasMaxLength(10);

            // Table & Column Mappings
            builder.ToTable("Person", "Person");
            builder.Property(t => t.BusinessEntityID).HasColumnName("BusinessEntityID");
            builder.Property(t => t.PersonType).HasColumnName("PersonType");
            builder.Property(t => t.NameStyle).HasColumnName("NameStyle");
            builder.Property(t => t.Title).HasColumnName("Title");
            builder.Property(t => t.FirstName).HasColumnName("FirstName");
            builder.Property(t => t.MiddleName).HasColumnName("MiddleName");
            builder.Property(t => t.LastName).HasColumnName("LastName");
            builder.Property(t => t.Suffix).HasColumnName("Suffix");
            builder.Property(t => t.EmailPromotion).HasColumnName("EmailPromotion");
            builder.Property(t => t.AdditionalContactInfo).HasColumnName("AdditionalContactInfo");
            builder.Property(t => t.rowguid).HasColumnName("rowguid");
            builder.Property(t => t.ModifiedDate).HasColumnName("ModifiedDate");

            builder.Ignore(t => t.PhoneNumbers);

            builder.HasOne(t => t.Employee).WithOne(t => t.PersonDetails).HasForeignKey<Employee>(p => p.BusinessEntityID);
            builder.HasOne(t => t.Password).WithOne(pw => pw.Person).HasForeignKey<Password>(pw => pw.BusinessEntityID);
        }
    }
}

File: Test/AdventureWorksLegacyModelVB/Mapping/PersonMap.vb
Imports System.Data.Entity.ModelConfiguration

Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Metadata.Builders
#Disable Warning BC8602

Namespace AW.Mapping
	Public Class PersonMap
		Inherits EntityTypeConfiguration(Of Person)

		Public Sub New()
			' Primary Key
			HasKey(Function(t) t.BusinessEntityID)

			[Property](Function(t) t.Title).HasMaxLength(8)

			[Property](Function(t) t.FirstName).IsRequired().HasMaxLength(50)

			[Property](Function(t) t.MiddleName).HasMaxLength(50)

			[Property](Function(t) t.LastName).IsRequired().HasMaxLength(50)

			[Property](Function(t) t.Suffix).HasMaxLength(10)

			' Table & Column Mappings
			ToTable("Person", "Person")
			[Property](Function(t) t.BusinessEntityID).HasColumnName("BusinessEntityID")
			[Property](Function(t) t.PersonType).HasColumnName("PersonType")
			[Property](Function(t) t.NameStyle).HasColumnName("NameStyle")
			[Property](Function(t) t.Title).HasColumnName("Title")
			[Property](Function(t) t.FirstName).HasColumnName("FirstName")
			[Property](Function(t) t.MiddleName).HasColumnName("MiddleName")
			[Property](Function(t) t.LastName).HasColumnName("LastName")
			[Property](Function(t) t.Suffix).HasColumnName("Suffix")
			[Property](Function(t) t.EmailPromotion).HasColumnName("EmailPromotion")
			[Property](Function(t) t.AdditionalContactInfo).HasColumnName("AdditionalContactInfo")
			[Property](Function(t) t.rowguid).HasColumnName("rowguid")
			[Property](Function(t) t.ModifiedDate).HasColumnName("ModifiedDate")

			HasOptional(Function(t) t.Employee).WithRequired(Function(t) t.PersonDetails)
			HasOptional(Function(t) t.Password).WithRequired(Function(pw) pw.Person)
		End Sub
	End Class

	Public Module Mapper
		<System.Runtime.CompilerServices.Extension> _
		Public Sub Map(ByVal builder As EntityTypeBuilder(Of Person))
			builder.Property(Function(t) t.Title).HasMaxLength(8)

			builder.Property(Function(t) t.FirstName).IsRequired().HasMaxLength(50)

			builder.Property(Function(t) t.MiddleName).HasMaxLength(50)

			builder.Property(Function(t) t.LastName).IsRequired().HasMaxLength(50)

			builder.Property(Function(t) t.Suffix).HasMaxLength(10)

			' Table & Column Mappings
			builder.ToTable("Person", "Person")
			builder.Property(Function(t) t.BusinessEntityID).HasColumnName("BusinessEntityID")
			builder.Property(Function(t) t.PersonType).HasColumnName("PersonType")
			builder.Property(Function(t) t.NameStyle).HasColumnName("NameStyle")
			builder.Property(Function(t) t.Title).HasColumnName("Title")
			builder.Property(Function(t) t.FirstName).HasColumnName("FirstName")
			builder.Property(Function(t) t.MiddleName).HasColumnName("MiddleName")
			builder.Property(Function(t) t.LastName).HasColumnName("LastName")
			builder.Property(Function(t) t.Suffix).HasColumnName("Suffix")
			builder.Property(Function(t) t.EmailPromotion).HasColumnName("EmailPromotion")
			builder.Property(Function(t) t.AdditionalContactInfo).HasColumnName("AdditionalContactInfo")
			builder.Property(Function(t) t.rowguid).HasColumnName("rowguid")
			builder.Property(Function(t) t.ModifiedDate).HasColumnName("ModifiedDate")

			builder.Ignore(Function(t) t.PhoneNumbers)

			builder.HasOne(Function(t) t.Employee).WithOne(Function(t) t.PersonDetails).HasForeignKey(Of Employee)(Function(p) p.BusinessEntityID)
			builder.HasOne(Function(t) t.Password).WithOne(Function(pw) pw.Person).HasForeignKey(Of Password)(Function(pw) pw.BusinessEntityID)
		End Sub
	End Module
End Namespace